### DISCUSSION TITLE
vue-google-maps code history

### DISCUSSION THREAD

**Discussion by aentwist** [NathanAP/vue-google-maps-community-fork#43] [General]
Perhaps the original project. Built for Vue 1.
https://github.com/GuillaumeLeclerc/vue-google-maps

A long-standing Vue 2 fork.
https://github.com/xkjyeah/vue-google-maps

The succeeding Vue 2 fork.
https://github.com/diegoazh/gmap-vue

And finally the fork for Vue 3, which broke the obvious fork history.
https://github.com/fawmi/vue-google-maps

And now this successor, I suppose.

As it ages, this whole thing *really* needs ported to TypeScript 🙂

**Comment by NathanAP** [comment]
Hello! We were talking about this in #39 . I don't have any experience with Typescript, but I'm 100% available to help in anyway if someone wants to do this. Also, we have a person that transformed this package into Typescript but didn't published it into npm yet, so maybe we could just wait for him to do it.

**Comment by davydnorris** [comment]
Hi all,
I was an active contributor on Diego's fork (diegoazh) and added the Heat map and Drawing Manager components. Diego then went quiet until very recently when he released his V3 version but, to be honest, I find it very over engineered and I don't believe the complexity is justified. So I've been looking for a library to migrate to that goes back to the simplicity of the original, and I think this is it. 

I also found a few things I really needed to change in my Drawing Manager and I've updated it, but now he's moved on to Vue3.

So I'm going to clone this and start re-implementing my stuff based on this. I do however have some suggestions around the use of shapes:
 - in Diego's port we found that there were some serious performance problems as the number of objects on the map increased
 - I believe this was, in part, due to the added code complexity, but I also believe that it's due to not being smart about the shape components
 - I was seeing event storms coming from shapes whenever they got moved on the map in some way, and I think we should tie the event binding of drag events and edit events to the editable and draggable props so that they only get enabled when a shape is marked as editable and/or draggable
 - I also think that perhaps only the drag start and end events should be emitted by default and a fine grained event prop be added to turn everything on. In addition I think that when the editable prop is true, the drag events should be silenced and perhaps vice versa
 - this is especially true for the Polyline type shapes, where events should be a lot smarter. You can drag a single point, or multiple points, or the entire polyshape. This happens during editing and you get both drag and edit events fired

**Reply by NathanAP** [reply]
Hello! Thanks for your concern about all this and thanks for sharing all the information! Feel free to share your migration, we're here to help if you need :)

I believe you should just take care about #15, its a PR that we could not find a good solution for
